spec/creator_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'creator'
+
+RSpec.describe Creator do
+  describe '#run' do
+    let(:database) { double('ArrayDatabase', create: nil) }
```

Comment:
In this case, the database is simple enough that I would use it directly:

```ruby
let(:database) { ArrayDatabase.new }
```

Try to use mocks only when needed. For example in the case of the user interface, you need it because you have to wait for user input and you don't have a human user entering things in automated tests.

Reply:
The reason I made a double of the database is to test create being called.
I don't think I can do that without making a double of some kind.

Reply:
you don't have to test that create was called
what does create do?
that is what you have to test :+1:

Reply:
"in this case....... that is what you have to test"
Does it take ages to learn all these rules? 
Seems like there are so many rules and with difficult correlations.